File: Recruit/RecruitClient/src/validations/models/candidate.js
const candidateSchema = {
  fullname: {
    type: 'string',
    maxLength: 50,
    minLength: 1,
  },
  email: {
    type: 'string',
    maxLength: 256,
    minLength: 1,
    pattern: '^[A-Za-z0-9_!#$%&*+/=?`\'{|}~^.-]+@[A-Za-z0-9.-]+$',
  },
  birthday: {
    type: 'string',
    minLength: 1,
    pattern: '^(\\d{4})\\/(0[1-9]|1[0-2])\\/(0[1-9]|[1-2][0-9]|3[0-1])$',
    dateMustBeInPast: true,
    checkDate: true,
  },
  start_join_date: {
    type: 'string',
    minLength: 1,
    pattern: '^(\\d{4})\\/(0[1-9]|1[0-2])\\/(0[1-9]|[1-2][0-9]|3[0-1])$',
    dateIsFuture: true,
  },
  full_address: {
    type: 'string',
    minLength: 1,
    maxLength: 256,
  },
  telephone_no: {
    type: 'string',
    minLength: 1,
    pattern: '^0(\\d{9,10})$',
  },
  cv_file: {
    type: 'object',
    properties: {
      file_size: {
        type: 'string',
      },
      file_ext: {
        type: 'string',
      },
      file: {
        type: 'string',
        minLength: 1,
        errorMessage: {
          minLength: 'Vui lòng chọn file',
          _: 'Vui lòng chọn file',
        },
      },
    },
  },
  team_id: {
    type: 'string',
    minLength: 1,
  },
  position_id: {
    type: 'string',
    minLength: 1,
  },
  office_id: {
    type: 'string',
    minLength: 1,
  },
  recommender_id: {
    type: 'string',
  },
  place_of_birth: {
    type: 'string',
    minLength: 1,
    maxLength: 256,
  },
  identification_number: {
    type: 'string',
    minLength: 1,
    pattern: '^[0-9]{12}$',
  },
  date_issued_identification: {
    type: 'string',
    minLength: 1,
    pattern: '^(\\d{4})\\/(0[1-9]|1[0-2])\\/(0[1-9]|[1-2][0-9]|3[0-1])$',
    dateMustBeInPast: true,
  },
  place_issued_identification: {
    type: 'string',
    minLength: 1,
    maxLength: 100,
  },
  bank_account: {
    type: 'string',
    maxLength: 50,
    pattern: '^\\d+$|^$',
  },
  bank_branch: {
    type: 'string',
    maxLength: 256,
  },
  vehicle_number: {
    type: 'string',
    maxLength: 9,
    pattern: '^[A-Za-z0-9]+$|^$',
  },
  gender: {
    type: 'string',
    pattern: '^[0-2]$',
  },
  number_experiences: {
    type: 'string',
    minLength: 1,
    pattern: '^\\d{1,5}$|(?=^.{1,5}$)^\\d+\\.\\d{0,1}$',
  },
};

export default candidateSchema;

File: Core/src/dto/response/candidates/candidate.py
"""
Candidate item response model
"""

from typing import Optional
from pydantic import BaseModel
from helpers.kbn import (CandidateStatus)


class CandidateItemResponse(BaseModel):
  """
    Candidate item model
  """
  id: int
  fullname: str
  email: str
  full_address: str
  place_of_birth = str
  telephone_no: str
  application_date: str
  position_id: int
  team_id: int
  cv_file_path: str
  recommender_id: Optional[int] = None
  status: CandidateStatus
  previous_status: Optional[CandidateStatus] = None
  office: str
  gender: str


class CandidateConfirmResponse(BaseModel):
  """
    Confirm candidate item model
  """
  id: int
  fullname: str
  team: Optional[str] = None
  position: Optional[str] = None
  email: str
  full_address: Optional[str] = None
  telephone_no: Optional[str] = None
  birthday: Optional[str] = None
  school: Optional[str] = None
  place_issued_identification: Optional[str] = None
  identification_number: Optional[str] = None
  date_issued_identification: Optional[str] = None
  start_join_date: Optional[str] = None
  class_room: Optional[str] = None
  bank_account: Optional[str] = None
  bank_branch: Optional[str] = None
  vehicle_number: Optional[str] = None
  department: Optional[str] = None
  place_of_birth: Optional[str] = None
